lib/goto-definition.coffee

```diff
@@ -66,7 +66,17 @@ module.exports =
     scan_regex = scan_regex.filter (e, i, arr) -> arr.lastIndexOf(e) is i
     scan_paths = scan_paths.filter (e, i, arr) -> arr.lastIndexOf(e) is i
 
-    regex = scan_regex.join('|').replace(/{word}/g, word)
+    # Word recognition doesn't always work perfectly (e.g. clojure deref with
```

Comment:
So that might work for clojure, where the usual case is only `@myval` to deref a val declared as `(def myval)` which you'd want to strip.... but it would become difficult with perl6 I think, since you could define `$x`, `@x`, `\x`, `%x` and `&x` to all have different values, and only `&x` could be defined _without_ the sigil (as `sub x`) (note that it could _also_ be declared as `my &x`)... I don't know how you'd be able to make it work in the general case. But maybe you can see something I can't here.